Remove OpenFaaS links from website (#777)
vmware-samples/vcenter-event-broker-appliance: docs/kb/use-functions.md

### Deployment Prerequisites

Before proceeding to deploy a function, you must have VMware Event Broker Appliance deployed and be able to SSH to the appliance or have access to the kubernetes configuration file (`/root/.kube/config`) locally on your desktop to authenticate.

### Function Deployment with Kubernetes Secrets

Most functions require configuration parameters to work correctly. A function author can hard code parameter values inside the function. However, any changes to the configuration parameters means the function's container image must be rebuilt. By using configuration parameters stored in a Kubernetes secret, a function can be reconfigured by simply changing the secret. 

> For most of this walk-through, the [`kn-ps-echo`](/examples) function from the examples folder is used. The `kn-ps-echo` function does not use secrets because the function simply echoes any event that it receives. 

Reviewer: nit: I'd completely remove echo from this section. After you restructure, we already cover echo in the section earlier when describing how to deploy a simple function.

> When working with secrets, this walk-through uses the [`kn-ps-email`](/examples) function from the examples folder.

Reviewer: question: does this point to the github examples folder or is this a link on the VEBA website?

Author: It links on the website

## Knative

For most this walk-through, the [`kn-ps-echo`](/examples) function from the examples folder is used. For working with secrets, the [`kn-ps-email`](/examples) function from the examples folder is used.

### Knative Prerequisites

> Before proceeding to deploy a function, you must have VMware Event Broker Appliance deployed and be able to SSH to the appliance or have access to the kubernetes configuration file (/root/.kube/config) locally on your desktop to authenticate

Reviewer: ```suggestion
Before proceeding to deploy a function, you must have VMware Event Broker Appliance deployed and be able to SSH to the appliance or have access to the kubernetes configuration file (`/root/.kube/config`) locally on your desktop to authenticate.
```

Author: Fixed

### Knative Function Deployment with Secrets

Most functions require configuration parameters to work correctly. A function author can hard code values inside the function. However, any changes to the configuration parameters means the function's container image must be rebuilt. By using configuration parameters stored in a Kubernetes secret, a function can be reconfigured by simply changing the secret.

> You can verify whether a function requires a secret by looking at the function's `function.yaml`. Below is part of the `function.yaml` for `kn-ps-email`. Note the `secretRef:` section specifying the name of the secret the function expects.

Reviewer: ```suggestion
You can verify whether a function requires a secret by looking at the function's `function.yaml`. The `kn-ps-echo` function does not use secrets because the function simply echoes any event that it receives.

Below is part of the `function.yaml` for `kn-ps-email`. Note the `secretRef:` section specifying the name of the secret the function expects.
```

Author: Moved

```yaml
      containers:
        - image: us.gcr.io/daisy-284300/veba/kn-ps-email:1.4
          envFrom:
            - secretRef:
                name: email-secret
          env:
            - name: FUNCTION_DEBUG
              value: "false"
```

> The `kn-ps-echo` function does not use secrets because the function simply echoes any event that it receives. The `kn-ps-email` function uses several configuration parameters. The secrets file for `kn-ps-email` is named `email_secrets.json`. 

Author: Moved

```json
{
    "SMTP_SERVER" : "smtp.primp-industries.com",
    "SMTP_PORT" : "587",
    "SMTP_USERNAME" : "email@primp-industries.com",
    "SMTP_PASSWORD" : "FILE-ME-IN-PLEASE",
    "EMAIL_SUBJECT" : "⚠️ [VM Delete Notification] ⚠️",
    "EMAIL_TO": "admins@primp-industries.com,sre@primp-industries.com",
    "EMAIL_FROM" : "notification-do-not-reply@primp-industries.com"
}
```
To create the Kubernetes secret via the command line, modify the secrets file with configuration data specific to your environment. Then run this command:
```bash
kubectl -n vmware-functions create secret generic email-secret --from-file=EMAIL_SECRET=email_secret.json
```

> `email-secret`, the argument just before `--from-file=`, is the name that Kubernetes uses to refer to the secret. This name must match the `secretRef` in `function.yaml`. `EMAIL_SECRET`, the value just after `--from-file=`, is the environment variable that the secrets get saved into. You can find the environment variable the function expects by looking at `handler.ps1`. 

Reviewer: ```suggestion
`email-secret`, the argument just before `--from-file=`, is the name that Kubernetes uses to refer to the secret. This name must match the `secretRef` in `function.yaml`. 

`EMAIL_SECRET`, the value just after `--from-file=`, is the environment variable that the secret gets loaded into the function. You can find the environment variable the function expects by looking at `handler.ps1`. 
```

```powershell
try {
    $jsonSecrets = ${env:EMAIL_SECRET} | ConvertFrom-Json
} catch {
    throw "`nK8s secrets `$env:EMAIL_SECRET does not look to be defined"
}
```

> If you deployed VEBA with the vCenter UI option, you can create the secret by clicking on the `Secrets` section, then clicking `Add`. The `Secret Name` must match the `secretRef` in `function.yaml`. The `Secret Key` must match the environment variable the function expects, as shown in `handler.ps1` above. The `Secret Value` is the JSON copied from `email_secret.json`, modified with configuration data specific to your environment. Once all fields are filled out, click `Add Secret`.

Author: Updated

    "VC_PASSWORD" : "FillMeIn"
}
```

> Then create the Kubernetes secret with this command:

Reviewer: IMHO the whole section on `secrets` need to be moved. Currently its listed under the verification section.

Suggestion: create a separate section "Knative Function Deployment with Secrets" and describe how to create a secret **before** deploying a function via `kubectl` or `UI`. Also advise on the change in `function.yaml` respectively steps in the UI to attach a secret.

Author: Updated
